simplify run_test for distributed tests

albanD opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73210
* __->__ #73209

Ensure it will be more future proof when more files get added.
Note that this adds back to this list one test that was missing.

albanD committed:
simplify run_test for distributed tests

[ghstack-poisoned]

malfet reviewed (approved):
Looks good to me, although please note it adds `test_nccl`:
```
 % python3 -c 'from run_test import TESTS, DISTRIBUTED_TESTS;print({test for test in TESTS if test.startswith("distributed")}.symmetric_difference(set(DISTRIBUTED_TESTS)))'
Ignoring disabled issues:  []
{'distributed/test_nccl'}
```

albanD commented:
Yes that's what I meant in the description by "adds one test that was missing"

albanD committed:
Update on "simplify run_test for distributed tests"


Ensure it will be more future proof when more files get added.
Note that this adds back to this list one test that was missing.

[ghstack-poisoned]

albanD commented:
@pytorchbot merge this please
